src/py/flwr/client/mod/centraldp_mods.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2024 Flower Labs GmbH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Clipping modifiers for central DP with client-side clipping."""
+
+
+from flwr.client.typing import ClientAppCallable
+from flwr.common import ndarrays_to_parameters, parameters_to_ndarrays
+from flwr.common import recordset_compat as compat
+from flwr.common.constant import MESSAGE_TYPE_FIT
+from flwr.common.context import Context
+from flwr.common.differential_privacy import compute_clip_model_update
+from flwr.common.differential_privacy_constants import KEY_CLIPPING_NORM
+from flwr.common.message import Message
+
+
+def fixedclipping_mod(
+    msg: Message, ctxt: Context, call_next: ClientAppCallable
+) -> Message:
+    """Client-side fixed clipping modifier.
+
+    This mod needs to be used with the DifferentialPrivacyClientSideFixedClipping
+    server-side strategy wrapper.
+
+    The wrapper sends the clipping_norm value to the client.
+
+    This mod clips the client model updates before sending them to the server.
+
+    It operates on messages with type MESSAGE_TYPE_FIT.
+
+    Note: Consider the order of mods when using multiple.
+
+    Typically, fixedclipping_mod should be the last to operate on params.
+    """
+    if msg.metadata.message_type == MESSAGE_TYPE_FIT:
```

Comment:
We can unindent all of the below by doing a `if not msg...: return call_next(...)` here